examples/anyons.jl

```diff
@@ -0,0 +1,33 @@
+# # Anyons
+
+# We solve the almost-bosonic anyon model of https://arxiv.org/pdf/1901.10739.pdf
+
+using DFTK
+using StaticArrays
+using Plots
+
+# Unit cell. Having one of the lattice vectors as zero means a 2D system
+a = 14
+lattice = a .* [[1 0 0.]; [0 1 0]; [0 0 0]];
+
+# Confining scalar potential
+pot(x, y, z) = ((x - a/2)^2 + (y - a/2)^2)
+
+# Parameters
+Ecut = 50
+n_electrons = 1
+β = 5
+
+terms = [Kinetic(2),
+         ExternalFromReal(X -> pot(X...)),
+         Anyonic(1, β)
+]
+model = Model(lattice; n_electrons=n_electrons,
+              terms=terms, spin_polarization=:spinless)  # "spinless electrons"
+basis = PlaneWaveBasis(model, Ecut, kgrid=(1, 1, 1); fft_size=DFTK.compute_fft_size(lattice, Ecut, supersampling=1.2))
+scfres = direct_minimization(basis, tol=1e-14)  # Reduce tol for production
+E = scfres.energies.total
+s = 2
+E11 = π/2 * (2(s+1)/s)^((s+2)/s) * (s/(s+2))^(2(s+1)/s) * E^((s+2)/s) / β
```

Comment:
This is super hard to read. Some more spaces could really help here.

Reply:
It's an ugly formula, spaces won't make it less ugly (in contrary, in my opinion...)

Reply:
I generally disagree. Spacing and subexpressions generally help. Just try a little to come up with something better.

Reply:
How would it help here? I'm copying the formula from https://arxiv.org/pdf/1901.10739.pdf, I think it's the nicest way to render this (particularly ugly and not meant to be read) formula

Reply:
```suggestion
ratio = (s+1) / s
rplus = (s+2) / s
E11 = π/2 * (2ratio)^rplus * rplus^(2ratio) * E^rplus / β
```

Reply:
I think that's much clearer.

Reply:
No way. These ratio and rplus names and the formula are actively misleading : they suggest meaning that just isn't there. It's just a stupid formula coming from a stupid integral. Again you're not really supposed to actually read this formula, it's just for checking that the code works.

Reply:
ok. I tried.